sunpy/net/helio/hec.py

```diff
@@ -156,39 +156,29 @@ def make_table_list(self):
 
         Returns
         -------
-        temp: `str`
+        `bytes`
             contains the name of the table that the user picked.
 
         Examples
         --------
         >>> from sunpy.net.helio import hec
         >>> hc = hec.HECClient()  # doctest: +REMOTE_DATA
-        >>> hc.make_table_list()  # doctest: +REMOTE_DATA +SKIP
+        >>> hc.select_table()  # doctest: +REMOTE_DATA +SKIP
 
         """
-        table_list = []
         tables = self.get_table_names()
-        for i in tables:
-            table = i[0]
-            if len(table) > 0:
-                table_list.append(table)
+        table_list = [t[0] for t in tables if len(t[0]) > 0]
         table_list.sort()
         for index, table in enumerate(table_list):
-            print(('{number:3d}) {table}'.format(number=index + 1,
-                                                 table=table)))
+            print(f'{index + 1} - {table.decode()}')
 
         while True:
-            stdinput = input("\nPlease enter a table number between 1 and "
-                             "{elem:d} "
-                             "('e' to exit): ".format(elem=len(table_list)))
-            if stdinput.lower() == "e" or stdinput.lower() == "exit":
-                temp = None
-                break
-            temp = [int(s) for s in stdinput.split() if s.isdigit()]
-            temp = temp[0] - 1
-            if temp in range(0, len(table_list)):
-                temp = table_list[temp]
-                break
```

Comment:
Do we not still need the break?

Reply:
We don't need them because return statements not only break out of the while loop, they also come out of function definition. They transfer control back to the statement which invoked the function.

Reply:
Fair enough.